--- a/diskcache/core.py	2026-03-05 15:53:58
+++ b/diskcache/core.py	2026-03-05 15:53:58
@@ -831,7 +831,6 @@
             ' expire_time = ?,'
             ' access_time = ?,'
             ' access_count = ?,'
-            ' tag = ?,'
             ' size = ?,'
             ' mode = ?,'
             ' filename = ?,'
@@ -842,7 +841,6 @@
                 expire_time,
                 now,  # access_time
                 0,  # access_count
-                tag,
                 size,
                 mode,
                 filename,
