--- a/lark/visitors.py
+++ b/lark/visitors.py
@@ -507,7 +507,7 @@
 def _vargs_meta(f, _data, children, meta):
     return f(meta, children)
 def _vargs_tree(f, data, children, meta):
-    return f(Tree(data, children, meta))
+    return f(Tree(data, list(reversed(children)), meta))


 def v_args(inline: bool = False, meta: bool = False, tree: bool = False, wrapper: Optional[Callable] = None) -> Callable[[_DECORATED], _DECORATED]:
