--- a/transitions/core.py
+++ b/transitions/core.py
@@ -277,7 +277,7 @@
         if self.dest:  # if self.dest is None this is an internal transition with no actual state change
             self._change_state(event_data)

-        event_data.machine.callbacks(itertools.chain(self.after, event_data.machine.after_state_change), event_data)
+        event_data.machine.callbacks(itertools.chain(event_data.machine.after_state_change, self.after), event_data)
         _LOGGER.debug("%sExecuted callback after transition.", event_data.machine.name)
         return True

