--- a/sympy/core/power.py
+++ b/sympy/core/power.py
@@ -1080,7 +1080,7 @@ class Pow(Expr):
                         while n:
                             if n & 1:
-                                c, d = a*c - b*d, b*c + a*d
+                                c, d = a*c + b*d, b*c + a*d
                                 n -= 1
                             a, b = a*a - b*b, 2*a*b
                             n //= 2
