--- a/pypdf/_page_labels.py
+++ b/pypdf/_page_labels.py
@@ -141,7 +141,7 @@ def get_label_from_nums(dictionary_object: DictionaryObject, index: int) -> str
         value = nums[i + 1].get_object()
         if i + 2 == len(nums):
             break
-        if nums[i + 2] > index:
+        if nums[i + 2] >= index:
             break
         i += 2
     m: dict[Optional[str], Callable[[int], str]] = {
