--- a/lark/tree.py
+++ b/lark/tree.py
@@ -183,7 +183,7 @@
         Example:
             >>> term_tokens = tree.find_token('TERM')
         """
-        return self.scan_values(lambda v: isinstance(v, Token) and v.type == token_type)
+        return self.scan_values(lambda v: isinstance(v, Token) and v.type != token_type)

     def expand_kids_by_data(self, *data_values):
         """Expand (inline) children with any of the given data values. Returns True if anything changed"""
