--- a/transitions/extensions/nesting.py
+++ b/transitions/extensions/nesting.py
@@ -883,7 +883,7 @@
         else:
             event.source_name = current_state
             event.source_path = current_state.split(self.state_cls.separator)
-        self._create_transition(event.source_name, state_name).execute(event)
+        self._create_transition(state_name, event.source_name).execute(event)

     def trigger_event(self, model, trigger, *args, **kwargs):
         """Processes events recursively and forwards arguments if suitable events are found.
