--- a/dateutil/parser/_parser.py
+++ b/dateutil/parser/_parser.py
@@ -550,7 +550,7 @@ class _ymd(list):
             else:
                 if (self[0] > 31 or
                     self.ystridx == 0 or
-                        (yearfirst and self[1] <= 12 and self[2] <= 31)):
+                        (yearfirst and self[1] > 12 and self[2] <= 31)):
                     # 99-01-01
                     if dayfirst and self[2] <= 12:
                         year, day, month = self
