--- a/babel/numbers.py
+++ b/babel/numbers.py
@@ -861,7 +861,7 @@
     # locale data. So there is no point implementing that.
     # Step 2.

     # Correct number to numeric type, important for looking up plural rules:
-    number_n = float(number) if isinstance(number, str) else number
+    number_n = float(number) if isinstance(number, str) else int(float(number))

     # Step 3.
     unit_pattern = get_currency_unit_pattern(currency, count=number_n, locale=locale)
