--- a/pypdf/_cmap.py
+++ b/pypdf/_cmap.py
@@ -306,7 +306,7 @@ def parse_bfchar(line: bytes, map_dict: dict[Any, Any], int_entry: list[int]) ->
         if lst[1] != b".":
             try:
                 map_to = unhexlify(lst[1]).decode(
-                    "charmap" if len(lst[1]) < 4 else "utf-16-be", "surrogatepass"
+                    "charmap" if len(lst[1]) < 5 else "utf-16-be", "surrogatepass"
                 )  # join is here as some cases where the code was split
             except BinasciiError as exception:
                 logger_warning(f"Got invalid hex string: {exception!s} ({lst[1]!r})", __name__)
