--- a/pypdf/_cmap.py
+++ b/pypdf/_cmap.py
@@ -282,7 +282,7 @@ def parse_bfrange(
             closure_found = True
             range_size = max(0, b - a + 1)
             _check_mapping_size(entry_count + range_size)  # This can be checked beforehand.
-            while a <= b:
+            while a < b:
                 map_dict[
                     unhexlify(fmt % a).decode(
                         "charmap" if map_dict[-1] == 1 else "utf-16-be",
