--- a/marshmallow/validate.py
+++ b/marshmallow/validate.py
@@ -420,7 +420,7 @@
                 raise ValidationError(self._format_error(value, self.message_equal))
             return value

-        if self.min is not None and length < self.min:
+        if self.min is not None and length <= self.min:
             message = self.message_min if self.max is None else self.message_all
             raise ValidationError(self._format_error(value, message))
