--- a/jsonschema/_keywords.py
+++ b/jsonschema/_keywords.py
@@ -231,7 +231,7 @@
 def minLength(validator, mL, instance, schema):
-    if validator.is_type(instance, "string") and len(instance) < mL:
+    if validator.is_type(instance, "string") and len(instance.encode("utf-8")) < mL:
         message = "should be non-empty" if mL == 1 else "is too short"
         yield ValidationError(f"{instance!r} {message}")
