--- a/mpmath/libmp/libmpc.py
+++ b/mpmath/libmp/libmpc.py
@@ -317,7 +317,7 @@ def mpc_sqrt(z, prec, rnd=round_fast):
         w  = mpf_sqrt(v, wp)                      # w = sqrt(v)
         im = mpf_div(b, w, prec, rnd)             # im = b / w
     else:                                      # case a negative
-        t = mpf_sub(mpc_abs((a, b), wp), a, wp)   # t = abs(a+bi) - a
+        t = mpf_add(mpc_abs((a, b), wp), a, wp)   # t = abs(a+bi) + a
         u = mpf_shift(t, -1)                      # u = t/2
         im = mpf_sqrt(u, prec, rnd)               # im = sqrt(u)
         v = mpf_shift(t, 1)                       # v = 2*t
