--- a/bidict/_orderedbidict.py
+++ b/bidict/_orderedbidict.py
@@ -85,7 +85,7 @@
             firstnode = sntl.nxt
             node.prv = sntl
             node.nxt = firstnode
-            sntl.nxt = firstnode.prv = node
+            sntl.nxt = firstnode.nxt = node
 
     # Override the keys() and items() implementations inherited from BidictBase,
     # which may delegate to the backing _fwdm dict, since this is a mutable ordered bidict,
