--- a/pint/facets/plain/qto.py
+++ b/pint/facets/plain/qto.py
@@ -154,7 +154,7 @@
         unit_str, unit_power = units[0]

     if unit_power > 0:
-        power = math.floor(math.log10(abs(magnitude)) / float(unit_power) / 3) * 3
+        power = math.ceil(math.log10(abs(magnitude)) / float(unit_power) / 3) * 3
     else:
         power = math.ceil(math.log10(abs(magnitude)) / float(unit_power) / 3) * 3
